app/components/Form/Field.tsx

```diff
@@ -102,55 +102,81 @@ export function createField(Component: ComponentType<any>, options?: Options) {
     } = field;
     const { error, submitError, warning, touched } = meta;
     const anyError = error || submitError;
-    const hasError = showErrors && touched && anyError && anyError.length > 0;
-    const hasWarning = showErrors && touched && warning && warning.length > 0;
-    const fieldName = input && input.name;
-    const content = (
-      <>
-        <Flex>
-          {label && (
-            <div className={cx(styles.label, labelClassName)}>{label}</div>
-          )}
-          {description && (
-            <Tooltip
+    const hasError = showErrors && touched && anyError?.length > 0;
+    const hasWarning = showErrors && touched && warning?.length > 0;
+    const fieldName = input?.name;
+    // CheckBox and RadioButton should have an inline label
+    const inlineLabel =
+      Component.name === 'CheckBox' || Component.name === 'RadioButton';
+
+    const labelComponent = (
+      <Flex>
+        {label && (
+          <div
+            style={{
+              cursor: inlineLabel ? 'pointer' : 'default',
+              fontSize: !inlineLabel ? 'var(--font-size-large)' : 'inherit',
+            }}
+            className={labelClassName}
+          >
+            {label}
+          </div>
+        )}
+        {description && (
+          <Tooltip
+            style={{
+              display: 'inline-block',
+            }}
+            content={description}
+          >
+            <div
               style={{
-                display: 'inline-block',
+                marginLeft: '10px',
               }}
-              content={description}
             >
-              <div
-                style={{
-                  marginLeft: '10px',
-                }}
-              >
-                <Icon size={32} name="help" />
-              </div>
-            </Tooltip>
-          )}
-          {required && <span className={styles.required}>*</span>}
-        </Flex>
-        <Component
-          {...input}
-          {...props}
-          onChange={(value) => {
-            input.onChange(value);
-            onChange?.(value);
-          }}
-          className={cx(
-            className,
-            hasWarning && styles.inputWithWarning,
-            hasError && styles.inputWithError
-          )}
-        />
+              <Icon size={32} name="help" />
+            </div>
+          </Tooltip>
+        )}
+        {required && <span className={styles.required}>*</span>}
+      </Flex>
+    );
+
+    const component = (
+      <Component
+        {...input}
+        {...props}
+        onChange={(value) => {
+          input.onChange(value);
+          onChange?.(value);
+        }}
+        className={cx(
+          className,
+          hasWarning && styles.inputWithWarning,
+          hasError && styles.inputWithError
+        )}
+      />
+    );
+
+    const content = inlineLabel ? (
+      <Flex gap={7}>
+        {component}
+        {labelComponent}
+      </Flex>
+    ) : (
+      <>
+        {labelComponent}
```

Comment:
Why are these in the opposite order as opposed to line 163-164?

Reply:
It's because I think it looks better to have the label on the right side.  It's also the most normal way to do it, ref.
the search result of "checkboxes":
<img width="1552" alt="image" src="https://user-images.githubusercontent.com/69514187/230669358-96d847bc-95c9-4cb0-92c8-eee4ab9b3ccc.png">

On line 163-164 the label is above the component, so that's why it comes before.

Reply:
Okay! Yeah, I totally agree with having the checkbox to the left and label to the right!